bundles/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/handler/IhcHandler.java

```diff
@@ -542,7 +542,7 @@ private void connect() throws IhcExecption {
                     conf.username);
             ihc = new IhcClient(conf.hostname, conf.username, conf.password, conf.timeout, conf.tlsVersion);
             ihc.openConnection();
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE,
```

Comment:
This should be set in `initialize()` before scheduling the job - just before this:
https://github.com/openhab/openhab-addons/blob/123982dcc6eaf399676040a3f6bc35b77eb39e41/bundles/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/handler/IhcHandler.java#L184-L187

(unrelated - the `if` condition can be removed)

Reply:
Connect method is called by the thing initialize method immediately (init delay is 0), so thing status will be change immediately. But as connect method is also called when the connection is reopen in any reason, thing status update should be. In this phase, connection to IHC controller is already open (communication works), but some initialisation is still pending which might take a while, so unknown state is better than offline.

Reply:
In case of reconnect, the status should already be OFFLINE, and it will then either stay OFFLINE or go back to ONLINE. It's not expected to return to UNKNOWN after this. This would also lead to an additional status update since it will probably change to ONLINE or OFFLINE quickly after that?

So I don't think any status update is needed here anymore, since `initialize()` now correctly sets the initial status.

Reply:
Unknown state is as valid as online or offline. See thing [status concept](https://www.openhab.org/docs/concepts/things.html#thing-status) and status transition diagram.

Reply:
Yes, it's valid, but why would you want this additional status update?

Reply:
Because connection is open and login is done, but some initialization is still required, which can take up to 30 seconds. Status is not anymore offline but it’s not online yet either, but commans may be handled already. So state is exactly what is defined for unknown state.

Reply:
@jlaur I removed additional thing status update to get this merged

Reply:
I would have approved accepting your explanation, and also because it was unchanged from previously. 🙂 However, I have posted a small comment for the latest commit. After that, let's get it merged!